apply_neighbourhood.json

```diff
@@ -0,0 +1,217 @@
+{
+    "id": "apply_neighbourhood",
+    "summary": "Apply a process to pixels in a n-dimensional neighbourhood",
+    "description": "Apply a focal process to a data cube. A focal process is a process that works on a 'neighbourhood' of pixels. The neighbourhood can extend into multiple dimensions. An overlap can be specified so that neighbourhoods can have overlapping boundaries. This allows for continuity of the output.\n\nThe process must not add new dimensions, or remove entire dimensions, but the result can use different dimension labels.\n\nFor the special case of 2D convolution, it is recommended to use ``apply_kernel()``.",
+    "categories": [
+        "cubes"
+    ],
+    "parameters": [
+        {
+            "name": "data",
+            "description": "A data cube.",
+            "schema": {
+                "type": "object",
+                "subtype": "raster-cube"
+            }
+        },
+        {
+            "name": "process",
+            "description": "Process to be applied on all neighbourhoods.",
+            "schema": {
+                "type": "object",
+                "subtype": "process-graph",
+                "parameters": [
+                    {
+                        "name": "data",
+                        "description": "A subset of the data cube as specified in `context` and `overlap`.",
+                        "schema": {
+                            "type": "object",
+                            "subtype": "raster-cube"
+                        }
+                    },
+                    {
+                        "name": "context",
+                        "description": "Additional data passed by the user.",
+                        "schema": {
+                            "description": "Any data type."
+                        },
+                        "optional": true,
+                        "default": null
+                    }
+                ]
+            }
+        },
+        {
+            "name": "size",
+            "description": "Neighbourhood sizes along each dimension.\nThis object maps dimension names to either a physical measure, (e.g. 100 m, 10 days), or pixels (e.g. 32 pixels).\nFor dimensions not specified, the default is to provide all values. This may result in an exception if the dimension is too large to be fully included.",
```

Comment:
Do you think we need to say the following?

>  This may result in an exception if the dimension is too large to be fully included.

This is also true for other processes, but we never specify this explicitly. So I think we can remove it (or if it should stay, specify a concrete exception).

Reply:
I would prefer having some kind of recommendation to limit dimensions that don't have a clearly finite number of values, like the bands dimension. But not sure how to word it right.

Reply:
@jdries Maybe something like: 

> Be aware that including all values from very large dimensions may not be processed at once.

Is a bit inspired from the *_binary processes.